src/azure-cli/azure/cli/command_modules/role/_params.py

```diff
@@ -205,6 +205,35 @@ def load_arguments(self, _):
         c.argument('create_cert', action='store_true', arg_group='keyCredentials', help='Create a self-signed certificate to use for the credential')
         c.argument('keyvault', arg_group='keyCredentials', help='Name or ID of a KeyVault to use for creating or retrieving certificates.')
 
+    for item in ['app', 'sp']:
+        with self.argument_context(f'ad {item} federated-credential') as c:
+            # TODO: We have to decide which name to use
+            #   --credential-id is consistent with API
+            #   --key-id is consistent with passwordCredentials and keyCredentials
+            c.argument('federated_identity_credential_id_or_name', options_list=['--credential-id', '--key-id'],
```

Comment:
We need to decide whether `show/update/delete` command should take `--credential-id` or `--key-id`:
- `--credential-id`: Consistent with REST API `federatedIdentityCredentialId`
- `--key-id`: Consistent with other credentials’ (`passwordCredentials`, `keyCredentials`) `keyId`

Reply:
Can it support both?

Reply:
Implementing both certainly causes unnecessary learning effort and complication.